infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogEntryWriter.java

```diff
@@ -97,21 +131,115 @@ private boolean processMsg(LogReplicationEntryMsg txMessage) {
+                        List<UUID> streamTags = new ArrayList<>();
+                        if (dataStreamToTagsMap.containsKey(streamId)) {
+                            streamTags.addAll(dataStreamToTagsMap.get(streamId));
+                        }
+                        txnContext.logUpdate(streamId, smrEntry, streamTags);
                     }
                 }
             }
             txnContext.commit();
-
             lastMsgTs = Math.max(entryTs, lastMsgTs);
-            return true;
+        }
+
+        if (handlePendingEntries) {
+            updateLogReplicationConfig();
+            this.streamsSet = new HashSet<>(config.getStreamsToReplicate());
+            this.dataStreamToTagsMap = new HashMap<>(config.getDataStreamToTagsMap());
+            clearNewStreamsWithLocalWrites();
+            if (!pendingEntries.isEmpty()) {
+                try (TxnContext txnContext = logReplicationMetadataManager.getTxnContext()) {
+                    pendingEntries.entrySet().removeIf(mapEntry -> {
+                        UUID streamId = mapEntry.getKey();
+                        boolean toRemove = streamsSet.contains(streamId);
+                        if (toRemove) {
+                            log.info("Handle pending entry for stream id: {}", streamId);
+                            for (SMREntry smrEntry : pendingEntries.get(streamId)) {
+                                List<UUID> streamTags = new ArrayList<>();
+                                if (dataStreamToTagsMap.containsKey(streamId)) {
+                                    streamTags.addAll(dataStreamToTagsMap.get(streamId));
+                                }
+                                txnContext.logUpdate(streamId, smrEntry, streamTags);
+                            }
+                        }
+                        return toRemove;
+                    });
+                    txnContext.commit();
+                }
+            }
+            handlePendingEntries = false;
+        }
+
+        return true;
+    }
+
+    private void updateLogReplicationConfig() {
+        CorfuTable<TableName, CorfuRecord<TableDescriptors, TableMetadata>>
+                registryTable = rt.getTableRegistry().getRegistryTable();
+        Set<UUID> updatedStreamSet = registryTable.entryStream()
+                .filter(entry -> entry.getValue().getMetadata().getTableOptions().getIsFederated())
+                .map(entry -> CorfuRuntime.getStreamID(getFullyQualifiedTableName(entry.getKey())))
+                .collect(Collectors.toSet());
+        Map<UUID, Set<UUID>> updatedStreamTagsMap = new HashMap<>();
+        registryTable.forEach((tableName, tableRecord) -> {
+            UUID streamId = CorfuRuntime.getStreamID(getFullyQualifiedTableName(tableName));
+            updatedStreamTagsMap.putIfAbsent(streamId, new HashSet<>());
+            updatedStreamTagsMap.get(streamId).addAll(
```

Comment:
Do we need to have the streamTags of all the streams in registryTable? Even for the ones which are not be replicated?